Foundation/Data.swift

```diff
@@ -1474,7 +1478,7 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
     /// - parameter buffer: The buffer of bytes to append. The size is calculated from `SourceType` and `buffer.count`.
     @inline(__always)
     public mutating func append<SourceType>(_ buffer : UnsafeBufferPointer<SourceType>) {
-        if buffer.isEmpty { return }
+        if buffer.count == 0 { return }
```

Comment:
This is wrong: https://github.com/apple/swift/blob/master/stdlib/public/SDK/Foundation/Data.swift#L1467

Reply:
Please refer to @ikesyo 's  comment related to PR https://github.com/apple/swift-corelibs-foundation/pull/1485/files.This change has been implemented as part of gardening work done by @ikesyo   and I have picked those changes from latest master of SCLF.

Reply:
:) ..Yeah anyway rebasing based on latest master